providers/default/ciphers/cipher_chacha20.h

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright 2019 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include "include/crypto/chacha.h"
+#include "internal/ciphers/ciphercommon.h"
+
+typedef struct {
+    union {
+        OSSL_UNION_ALIGN;  /* this ensures even sizeof(EVP_CHACHA_KEY)%8==0 */
```

Comment:
The comment here is misleading.  It doesn't guarantee anything about the `sizeof(EVP_CHACHA_KEY)`.   It ensures a natural alignment for the _d_ array.  It doesn't guarantee an eight byte alignment either (although it will on most systems).